Cabal/Distribution/Simple/GHC/Internal.hs

```diff
@@ -319,6 +319,7 @@ componentGhcOptions verbosity implInfo lbi bi clbi odir =
       ghcOptWarnMissingHomeModules = toFlag $ flagWarnMissingHomeModules implInfo,
       ghcOptPackageDBs      = withPackageDB lbi,
       ghcOptPackages        = toNubListR $ mkGhcOptPackages clbi,
+      ghcOptSplitSections   = toFlag (splitSections lbi),
       ghcOptSplitObjs       = toFlag (splitObjs lbi),
```

Comment:
One potential refactoring would be to fold these two fields into one `Maybe CodeShrinkScheme` with,
```haskell
data CodeShrinkScheme = SplitObjs | SplitSections
```
I'm not sure this is worth the trouble though.

Reply:
Yeah, don't bother. IMO `-split-objs` should be removed at some point.